okio/src/commonMain/kotlin/okio/FileSystem.kt

```diff
@@ -164,6 +185,26 @@ abstract class FileSystem {
   @Throws(IOException::class)
   abstract fun sink(file: Path): Sink
 
+  /**
+   * Creates a sink to write [file], executes [writerAction] to write it, and then closes the sink.
+   * This is a compact way to write a file.
+   */
+  inline fun <T> write(file: Path, writerAction: BufferedSink.() -> T): T {
+    return sink(file).buffer().use {
+      it.writerAction()
+    }
+  }
+
+  /**
+   * Creates a sink to write [file], executes [writerAction] to write it, and then closes the sink.
+   * This is a compact way to write a file.
+   *
+   * @param file the relative or absolute [path][Path] of the file to write.
+   */
+  inline fun <T> write(file: String, writerAction: BufferedSink.() -> T): T {
```

Comment:
This is the first time we’re coercing a string to a file in our APIs. I’m not 100% sure this is a good idea.

Reply:
Path is a better abstraction of filenames than Strings.  But String is the expedient default, two downsides to supporting string here are

1) You weaken the consistency of Path as the filename abstraction. People will use it with confusion.
2) You are opting in to build APIs that are not responsible for the edge cases around conversion. It seems cleaner for the caller to deal with that (or choose not to) outside the API.

Reply:
Lemme pull it out!